Code/src/main/java/nl/utwente/group10/haskell/type/VarT.java

```diff
@@ -49,6 +60,35 @@ public final String getName() {
     }
 
     /**
+     * @return Whether this type has any constraints.
+     */
+    public final boolean hasConstraints() {
+        return !this.constraints.isEmpty();
+    }
+
+    /**
+     * Checks whether the given type is within the constraints. If the set of constraints is empty, every type is within
+     * the constraints.
+     * @param type The type to check.
+     * @return Whether the given type is within the constraints of this type.
+     */
+    public final boolean hasConstraint(Type type) {
+        boolean out = false;
+
+        if (!this.constraints.isEmpty()) {
+            for (TypeClass typeclass : this.constraints) {
+                if (typeclass.hasType(type)) {
+                    out = true;
+                }
+            }
+        } else {
+            out = true;
+        }
+
+        return out;
```

Comment:
I'd be in favor of dropping the one-return rule here and going (psuedocode 'cause I'm lazy):

````
for (tc : this.c) {
  if (tc.hasType(type)) {
    return true;
  }
}

return false;
````